### PR TITLE
Created a widget for properties view in react-components

### PR DESCRIPTION
**Pull request by ngondalia**
Changes made:

- Implemented a properties widget in react-components
- Change theia implementation to use the widget from react-components

This will move the properties widget to the react-components npm package, so it can be used by both the theia extension and vscode extension.

Signed-off-by: Neel Gondalia ngondalia@blackberry.com

### CODE REVIEW

File: theia-extensions/viewer-prototype/src/browser/trace-explorer/trace-explorer-sub-widgets/trace-explorer-tooltip-widget.tsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
-import { ReactWidget, Message } from '@theia/core/lib/browser';
+import { Message, ReactWidget, Widget } from '@theia/core/lib/browser';
 import * as React from 'react';
 import { EditorOpenerOptions, EditorManager } from '@theia/editor/lib/browser';
 import URI from '@theia/core/lib/common/uri';
-import { Signals, signalManager } from 'traceviewer-base/lib/signals/signal-manager';
+import { ReactItemPropertiesWidget} from 'traceviewer-react-components/lib/trace-explorer/trace-explorer-tooltip-widget';
 
 @injectable()
 export class TraceExplorerTooltipWidget extends ReactWidget {
```
Comment (bhufmann) [inline theia-extensions/viewer-prototype/src/browser/trace-explorer/trace-explorer-sub-widgets/trace-explorer-tooltip-widget.tsx:9]: This class should be renamed to `TraceExplorerItemPropertiesWidget`

**Review by bhufmann — CHANGES_REQUESTED** [review]
Thanks for the contribution. It works well. 

I'd like to ask you though to update some names. I think it makes sense to do it now because before it becomes API that is used with Theia and VsCode. 

Basically, I'd like that tooltip is replaced with item properties where it makes sense. I tried to find the relevant places that need to be updated to make it easier. Refactoring of you code editor should make it easier. Add the renaming as a second commit to the PR. 

- in SignalManager.tsx: rename signal name and fireTooltipSignal(tooltip) to 
```
ITEM_PROPERTIES_UPDATED: 'item properties updated',
```
```
fireItemPropertiesSignalUpdated(properties?: { [key: string]: string; }): void {
        this.emit(Signals.ITEM_PROPERTIES_UPDATED, properties);
    }
- under `packages/...' directory: `trace-explorer-tooltip-widget.tsx` -> `trace-explorer-properties-widget.tsx`
- under `theia-extension/viewer-prototype/...` directory: `trace-explorer-tooltip-widget.tsx` -> `theia-trace-explorer-properties-widget.tsx`
```
- under `packages/react-compenents/src/trace-explorer`: rename `trace-explorer-tooltip-widget.tsx` to `trace-explorer-properties-widget.tsx`
- under `theia-extension/...`: rename `trace-explorer-tooltip-widget.tsx` to `theia-trace-explorer-properties-widget.tsx`

**Review by bhufmann — APPROVED** [review]
Looks good to me! Thank you for the contribution.

**Review by Rodrigoplp-work — APPROVED** [review]
Thank you for the contribution.

**Pull request merged**